### PR TITLE
Add resource group support

### PR DESCRIPTION
**Pull request by nolouch**

### CODE REVIEW

File: proto/resource_manager.proto [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+service ResourceManager {
```
Comment (nolouch) [inline proto/resource_manager.proto:12]: cc @sunxiaoguang @coocood

File: proto/resource_manager.proto [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  rpc ListResourceGroups(ListResourceGroupsRequest) returns (ListResourceGroupsResponse) {}
+
+  rpc GetResourceGroup(GetResourceGroupRequest) returns (GetResourceGroupResponse) {}
+
+  rpc AddResourceGroup(PutResourceGroupRequest) returns (PutResourceGroupResponse) {}
+
+  rpc ModifyResourceGroup(PutResourceGroupRequest) returns (PutResourceGroupResponse) {}
```
Comment (nolouch) [inline proto/resource_manager.proto:19]: cc @glorv

**Review by BornChanger — APPROVED** [review]
LGTM~

**Comment by BornChanger** [comment]
To confirm, will the http api be reserved or not?

File: pkg/kvrpcpb/kvrpcpb.pb.go [hunk 3]
```diff
@@ -3485,7 +3486,7 @@ type FlashbackToVersionRequest struct {
 	Version  uint64 `protobuf:"varint,2,opt,name=version,proto3" json:"version,omitempty"`
 	StartKey []byte `protobuf:"bytes,3,opt,name=start_key,json=startKey,proto3" json:"start_key,omitempty"`
 	EndKey   []byte `protobuf:"bytes,4,opt,name=end_key,json=endKey,proto3" json:"end_key,omitempty"`
-	// The `start_ts`` and `commit_ts` which the newly written MVCC version will use.
+	// The `start_ts“ and `commit_ts` which the newly written MVCC version will use.
```
Comment (BornChanger) [inline pkg/kvrpcpb/kvrpcpb.pb.go:3489]: What's the change for? And please fix the conflict also.

Comment (nolouch) [reply pkg/kvrpcpb/kvrpcpb.pb.go:3489]: fixed

**Comment by nolouch** [comment]
ptal @BornChanger

**Review by CabinfeverB — APPROVED** [review]

File: proto/resource_manager.proto [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  Error error = 1;
+  string body = 2;
+}
+
+message TokenBucketsRequest {
+  repeated TokenBucketRequst requests = 1;
+  uint64 target_request_period_ms = 2;
+}
+
+message TokenBucketRequst {
```
Comment (sunxiaoguang) [inline proto/resource_manager.proto:65]: There is a typo here. It should be `TokenBucketRequest`

**Pull request merged**